### PR TITLE
IOBJ fix - issue 3105

### LINKED ISSUE #3105: SAP BW component objects do not exist in SAP BASIS
Newly added class zcl_abapgit_object_iobj has references to data elements and function modules with prefix RSD_*. These do not exist in SAP BASIS so you cannot use the zabapgit report there now without activating with errors.

### PR DESCRIPTION
**Pull request by pawelwiejkut**
Fix for IOBJ issue #3105

### CODE REVIEW

File: src/objects/zcl_abapgit_object_iobj.clas.abap [hunk 1]
```diff
@@ -15,26 +15,46 @@ CLASS zcl_abapgit_object_iobj IMPLEMENTATION.
...
 
+    FIELD-SYMBOLS:
+      <lv_tstpnm> TYPE any,
+      <ls_viobj>  TYPE any.
+
+    lv_type  = 'rsd_s_viobj'.
     lv_objna = ms_item-obj_name.
 
+    TRY.
+        CREATE DATA lr_viobj TYPE (lv_type).
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_iobj.clas.abap]: the variable lv_type can be removed by doing

```suggestion
        CREATE DATA lr_viobj TYPE ('RSD_S_VI_OBJ').
```

Comment (pawelwiejkut) [reply src/objects/zcl_abapgit_object_iobj.clas.abap]: Done

File: src/objects/zcl_abapgit_object_iobj.clas.abap [hunk 2]
```diff
@@ -89,7 +107,25 @@ CLASS zcl_abapgit_object_iobj IMPLEMENTATION.
     DATA: ls_details TYPE bapi6108,
           lt_infoobj TYPE STANDARD TABLE OF bapi6108io,
           ls_return  TYPE bapiret2,
-          lt_return  TYPE STANDARD TABLE OF bapiret2.
+          lt_return  TYPE STANDARD TABLE OF bapiret2,
+          lt_comptab TYPE STANDARD TABLE OF spam_cvers.
+
+    CALL FUNCTION 'OCS_GET_INSTALLED_COMPS'
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_iobj.clas.abap]: suggest not adding this part, just try executing the code, if it fails then throw an error

File: src/objects/zcl_abapgit_object_iobj.clas.abap [hunk 3]
```diff
@@ -155,19 +191,33 @@ CLASS zcl_abapgit_object_iobj IMPLEMENTATION.
...
         i_iobjnm  = lv_objna
         i_objvers = 'A'
       IMPORTING
-        e_s_viobj = ls_viobj.
+        e_s_viobj = <ls_viobj>.
+
+    ASSIGN COMPONENT 'OBJSTAT' OF STRUCTURE <ls_viobj> TO <lv_objstat>.
 
-    IF ls_viobj-objstat = 'ACT'.
+    IF <lv_objstat> = 'ACT'.
```
Comment (larshp) [inline src/objects/zcl_abapgit_object_iobj.clas.abap]: also check `sy-subrc`

Comment (pawelwiejkut) [reply src/objects/zcl_abapgit_object_iobj.clas.abap]: Done

File: src/objects/zcl_abapgit_object_iobj.clas.abap (hunk 2 above)
Comment (pawelwiejkut) [reply src/objects/zcl_abapgit_object_iobj.clas.abap]: Done

**Comment by larshp** [comment]
I'm not exactly sure if `cx_sy_dyn_call_illegal_func` is thrown, but any how 👍

**Pull request merged**